File: mg_custom_nodes/Aaalice233_ComfyUI-Danbooru-Gallery_20260101/js/global/debug_config.js
/**
 * Debug配置管理模块 - 前端版本
 * 统一管理所有前端组件的debug模式开关
 */

import { createLogger } from '../global/logger_client.js';

// 创建logger实例
const logger = createLogger('debug_config');



class DebugConfig {
    constructor() {
        this.config = {
            group_executor_manager: false,
            group_executor_trigger: false,
            image_cache_save: false,
            image_cache_get: false,
            execution_engine: false,
            cache_control_events: false
        };
        this.loaded = false;
    }

    /**
     * 从后端API加载配置
     */
    async loadConfig() {
        try {
            const response = await fetch('/danbooru_gallery/get_debug_config');
            if (response.ok) {
                const data = await response.json();
                if (data.status === 'success') {
                    this.config = data.debug || this.config;
                    this.loaded = true;
                    logger.info('[DebugConfig] 配置加载成功:', this.config);
                    return true;
                }
            }
            logger.warn('[DebugConfig] 配置加载失败，使用默认配置');
            return false;
        } catch (error) {
            logger.error('[DebugConfig] 加载配置时出错:', error);
            return false;
        }
    }

    /**
     * 检查指定组件是否应该打印debug日志
     * @param {string} component - 组件名称
     * @returns {boolean} - true表示应该打印日志
     */
    shouldDebug(component) {
        return this.config[component] || false;
    }

    /**
     * 条件打印日志
     * @param {string} component - 组件名称
     * @param {...any} args - 要打印的内容
     */
    debugLog(component, ...args) {
        if (this.shouldDebug(component)) {
            logger.info(...args);
        }
    }

    /**
     * 条件打印警告
     * @param {string} component - 组件名称
     * @param {...any} args - 要打印的内容
     */
    debugWarn(component, ...args) {
        if (this.shouldDebug(component)) {
            logger.warn(...args);
        }
    }

    /**
     * 条件打印错误
     * @param {string} component - 组件名称
     * @param {...any} args - 要打印的内容
     */
    debugError(component, ...args) {
        if (this.shouldDebug(component)) {
            logger.error(...args);
        }
    }

    /**
     * 条件打印信息
     * @param {string} component - 组件名称
     * @param {...any} args - 要打印的内容
     */
    debugInfo(component, ...args) {
        if (this.shouldDebug(component)) {
            console.info(...args);
        }
    }

    /**
     * 获取所有配置
     * @returns {Object} - 配置对象
     */
    getAllConfig() {
        return { ...this.config };
    }

    /**
     * 更新配置（可选功能）
     * @param {Object} newConfig - 新的配置
     */
    async updateConfig(newConfig) {
        try {
            const response = await fetch('/danbooru_gallery/update_debug_config', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({ debug: newConfig })
            });

            if (response.ok) {
                const data = await response.json();
                if (data.status === 'success') {
                    this.config = newConfig;
                    logger.info('[DebugConfig] 配置更新成功');
                    return true;
                }
            }
            logger.error('[DebugConfig] 配置更新失败');
            return false;
        } catch (error) {
            logger.error('[DebugConfig] 更新配置时出错:', error);
            return false;
        }
    }
}

// 创建全局实例
window.debugConfig = new DebugConfig();

// 自动加载配置
window.debugConfig.loadConfig().then(() => {
    logger.info('[DebugConfig] Debug配置系统已初始化');
});

// 导出便捷函数
window.shouldDebug = (component) => window.debugConfig.shouldDebug(component);
window.debugLog = (component, ...args) => window.debugConfig.debugLog(component, ...args);
window.debugWarn = (component, ...args) => window.debugConfig.debugWarn(component, ...args);
window.debugError = (component, ...args) => window.debugConfig.debugError(component, ...args);
window.debugInfo = (component, ...args) => window.debugConfig.debugInfo(component, ...args);

File: mg_custom_nodes/Aaalice233_ComfyUI-Danbooru-Gallery_20260205/py/utils/debug_config.py
"""
Debug配置管理模块
统一管理所有组件的debug模式开关
"""

import json
import os
from typing import Dict, Any

# Logger导入
from .logger import get_logger
logger = get_logger(__name__)

# 配置文件路径（指向项目根目录的config.json）
CONFIG_FILE = os.path.join(os.path.dirname(os.path.dirname(__file__)), "config.json")

# 全局配置缓存
_debug_config: Dict[str, Any] = {}


def load_config() -> Dict[str, Any]:
    """加载配置文件"""
    global _debug_config

    try:
        if os.path.exists(CONFIG_FILE):
            with open(CONFIG_FILE, 'r', encoding='utf-8') as f:
                config = json.load(f)
                _debug_config = config.get('debug', {})
                return _debug_config
        else:
            # 如果配置文件不存在，使用默认配置（全部关闭）
            _debug_config = {
                "group_executor_manager": False,
                "group_executor_trigger": False,
                "image_cache_save": False,
                "image_cache_get": False,
                "execution_engine": False,
                "cache_control_events": False
            }
            return _debug_config
    except Exception as e:
        logger.error(f"加载配置文件失败: {str(e)}")
        # 发生错误时，默认关闭所有debug
        _debug_config = {}
        return _debug_config


def save_config(debug_config: Dict[str, Any]) -> bool:
    """保存配置文件"""
    global _debug_config

    try:
        # 读取现有配置
        existing_config = {}
        if os.path.exists(CONFIG_FILE):
            with open(CONFIG_FILE, 'r', encoding='utf-8') as f:
                existing_config = json.load(f)

        # 更新debug配置
        existing_config['debug'] = debug_config

        # 写回文件
        with open(CONFIG_FILE, 'w', encoding='utf-8') as f:
            json.dump(existing_config, f, indent=2, ensure_ascii=False)

        # 更新缓存
        _debug_config = debug_config

        logger.info(f"配置已保存")
        return True
    except Exception as e:
        logger.error(f"保存配置文件失败: {str(e)}")
        return False


def should_debug(component: str) -> bool:
    """
    检查指定组件是否应该打印debug日志

    Args:
        component: 组件名称，如 'group_executor_manager'

    Returns:
        bool: True表示应该打印日志，False表示不打印
    """
    # 如果配置未加载，先加载
    if not _debug_config:
        load_config()

    # 返回该组件的debug开关，默认为False
    return _debug_config.get(component, False)


def debug_print(component: str, *args, **kwargs):
    """
    条件打印日志，只有当组件的debug模式开启时才打印

    Args:
        component: 组件名称
        *args: 要打印的内容
        **kwargs: print函数的其他参数
    """
    if should_debug(component):
        # 强制刷新缓冲区，确保日志立即写入文件
        print(*args, **kwargs, flush=True)


def get_all_debug_config() -> Dict[str, Any]:
    """获取所有debug配置"""
    if not _debug_config:
        load_config()
    return _debug_config.copy()


# 初始化时加载配置
load_config()
